projects/testing-library/src/lib/testing-library.ts

```diff
@@ -133,12 +133,7 @@ export async function render<SutType, WrapperType = SutType>(
     >,
   ) => {
     const newComponentInputs = properties?.componentInputs ?? {};
-    for (const inputKey of renderedInputKeys) {
-      if (!Object.prototype.hasOwnProperty.call(newComponentInputs, inputKey)) {
-        delete (fixture.componentInstance as any)[inputKey];
-      }
-    }
-    setComponentInputs(fixture, newComponentInputs);
+    const changesInComponentInput = update(fixture, renderedInputKeys, newComponentInputs, setComponentInputs);
     renderedInputKeys = Object.keys(newComponentInputs);
 
     const newComponentOutputs = properties?.componentOutputs ?? {};
```

Comment:
On a quick local test, changes in component outputs are not included in ngOnChanges. So I think we only need to calculate changes from component inputs and component properties.

Reply:
Sounds good to me!
Thanks a lot for this, I was just going to take a stab at it but you beat me to it 👍